driver/compile_common.ml

```diff
@@ -23,10 +23,11 @@ type info = {
   native : bool;
 }
 
-let with_info ~native ~tool_name ~source_file ~output_prefix ~dump_ext k =
+let with_info
+    ~native ~tool_name ~source_file ~source_kind ~output_prefix ~dump_ext k =
```

Comment:
I have a small interrogation, with the new `source_kind` this is the third argument that we pass to `with_info` which is only used to construct an `Unit_info.t`. Do we want to group together those arguments in a single `unit_info` argument.

Reply:
That seems reasonable, I will give it a go, we can always revert later if we're not satisfied.

Reply:
Done in [2f2b546](https://github.com/ocaml/ocaml/pull/13286/commits/2f2b546c5779246771dd72dabfd0dff591c23a4e)